Formula/lib/libmsquic.rb

```diff
@@ -0,0 +1,50 @@
+class Libmsquic < Formula
+  desc "Cross-platform, C implementation of the IETF QUIC protocol"
+  homepage "https://github.com/microsoft/msquic"
+  url "https://github.com/microsoft/msquic.git",
+  tag:      "v2.4.4",
+  revision: "e87ae9a37bcebdf5ab3725e5d882ddb50e4dbbb4"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "openssl@3"
+
+  def install
+    args = %w[
+      -DQUIC_TLS=openssl3
+      -DQUIC_USE_SYSTEM_LIBCRYPTO=true
+      -DQUIC_BUILD_PERF=off
+      -DQUIC_BUILD_TOOLS=off
+      -DQUIC_BUILD_TESTS=off
+      -DHOMEBREW_ALLOW_FETCHCONTENT=on
```

Comment:
We can't avoid this right now, this requires upstream changes in the project and not sure if msquic is willing to do that.
`FetchContent` is only used to Fetch `quictls` here:
https://github.com/microsoft/msquic/blob/a09a6de23d724dc7cae9b5bf9dfb82205331a59a/CMakeLists.txt#L662-L678

Reply:
Per CMake documentation:
> If the dependency has already been populated earlier in this run, set the <lowercaseName>_POPULATED, <lowercaseName>_SOURCE_DIR, and <lowercaseName>_BINARY_DIR variables in the same way as a call to [FetchContent_GetProperties()](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_getproperties), then skip the remaining steps below and move on to the next dependency in the list.

https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable

Reply:
Btw, note that I'm just a bystander and not a Homebrew maintainer, better ask @carlocab if this FetchContent thing is important for Homebrew

Reply:
> 
> Can we use this to avoid fetching submodules again?

We definitely can, but not sure what will be the response from msquic to this, even if we can make it happen on main, it's highly unlikely that it's going to get backported to 2.4 release and make a new release for this.

Reply:
No, I mean this variables can be set during CMake configuration step, without updating the sources

Reply:
> No, I mean these variables can be set during CMake configuration step, without updating the sources

Silly me, you're right I'll give it a try!

Reply:
Unfortunately, no, we can't avoid it without changing source code, because `FetchContent` is getting available through `QUIC_TLS` option and `QUIC_TLS` is must, because it also configures some other stuff like `src/platform/CMakeLists.txt:59`.

Reply:
I don't see any contradictions, `QUIC_TLS` should be set, CMake interpreter goes in the `if(QUIC_TLS STREQUAL "openssl" OR QUIC_TLS STREQUAL "openssl3")` condition, but does not populate the submodule because of the variables you provided

You still need to provide `QUIC_TLS`, and CMake should stumble upon FetchContent, but consider it already populated

_Theoretically_

Reply:
Theoretically yes, but after I set those variables as populated I still got the same error as
```
Refusing to populate dependency 'OpenSSLQuic' with FetchContent while
  building in Homebrew, please use a formula dependency or add a resource to
  the formula.
```

Reply:
I don't know what brew maintainers will say, but I suggest leaving `HOMEBREW_ALLOW_FETCHCONTENT=ON` but still  using `<lowercaseName>_POPULATED`, `<lowercaseName>_SOURCE_DIR`, and `<lowercaseName>_BINARY_DIR` so `FetchContent` doesn't download anything

Looks like brew's trapper was not ready for this scenario

@carlocab what do you think?

Reply:
I can do that, but here is what's actually happening (if I understood correctly) submodules are already getting downloaded as part of `git submodules`, and we're not downloading anything with `FetchContent`, what `FetchContent` doing is just building `quictls` and make it a `usable library` with `add_library`. See:
https://github.com/microsoft/msquic/blob/a09a6de23d724dc7cae9b5bf9dfb82205331a59a/submodules/CMakeLists.txt#L10

Reply:
I found similar case: https://github.com/Homebrew/homebrew-core/blob/4b170e682e2950ca4ab38afd92b55ed26cb0e55b/Formula/m/mlx.rb#L74-L82
and setting these two parameters 
`
-DFETCHCONTENT_FULLY_DISCONNECTED=ON
-DFETCHCONTENT_TRY_FIND_PACKAGE_MODE=ALWAYS
`

along with `-DHOMEBREW_ALLOW_FETCHCONTENT=ON`, should do what we want, if I understood correctly, it will block any download over FetchContent.